src/components/Popup/Popup.ts

```diff
@@ -0,0 +1,145 @@
+ *
+ * @example
+ * Basic popup with dialog and ribbon content:
+ * ```html
+ * <nosto-popup name="promo-popup" segment="5b71f1500000000000000006">
+ *   <div slot="default">
+ *     <h2>Special Offer!</h2>
+ *     <p>Get 20% off your order today</p>
+ *     <button n-close>Close</button>
+ *   </div>
+ *   <div slot="ribbon">
+ *     <span>Limited time!</span>
+ *   </div>
+ * </nosto-popup>
+ * ```
+ */
+@customElement("nosto-popup")
+export class Popup extends NostoElement {
+  /** @private */
+  static attributes = {
+    name: String,
+    segment: String
+  }
+
+  name!: string
+  segment?: string
+
+  constructor() {
+    super()
+    this.attachShadow({ mode: "open" })
+  }
+
+  async connectedCallback() {
+    assertRequired(this, "name")
+
+    if (!(await isPopupShown(this))) {
+      this.style.display = "none"
+      return
+    }
+
+    const state = getPopupState(this.name)
+    const mode = state === "ribbon" ? "ribbon" : "open"
+    renderShadowContent(this, mode)
+    this.addEventListener("click", this.handleClick)
+  }
+
+  private handleClick = (event: Event) => {
+    const target = event.target as HTMLElement
+    const closeElement = target?.closest("[n-close]")
+    const ribbonElement = target?.closest("[n-ribbon]")
+
+    if (closeElement) {
+      event.preventDefault()
+      event.stopPropagation()
+      closePopup(this)
+    } else if (ribbonElement) {
+      event.preventDefault()
+      event.stopPropagation()
+      setPopupToRibbon(this.name)
+      renderShadowContent(this, "ribbon")
+    }
+  }
+}
+
+function renderShadowContent(element: Popup, mode: "open" | "ribbon" = "open") {
+  if (!element.shadowRoot) return
+
+  const dialogHidden = mode === "ribbon" ? "hidden" : ""
+  const ribbonHidden = mode === "open" ? "hidden" : ""
+
+  element.shadowRoot.innerHTML = `
+    <style>${popupStyles}</style>
+    <dialog open part="dialog" class="${dialogHidden}">
+      <slot name="default"></slot>
+    </dialog>
+    <div class="ribbon ${ribbonHidden}" part="ribbon">
+      <slot name="ribbon"></slot>
+    </div>
+  `
+}
+
+function closePopup(element: Popup) {
+  if (element.name) {
+    setPopupClosed(element.name)
+  }
+  element.style.display = "none"
+}
+
+async function isPopupShown(element: Popup): Promise<boolean> {
+  // Check if popup was permanently closed
+  const state = getPopupState(element.name)
+  if (state === "closed") {
+    return false
+  }
+
+  // Check segment precondition if specified
+  if (element.segment && !(await checkSegment(element.segment))) {
+    return false
+  }
+
+  return true
+}
+
+function getPopupState(name: string): "open" | "ribbon" | "closed" {
+  const key = `nosto:web-components:popup:${name}`
+  const state = localStorage.getItem(key)
+  if (state === "closed" || state === "ribbon") {
+    return state
+  }
+  return "open"
+}
+
+function setPopupClosed(name: string) {
```

Comment:
combine setPopupClosed and setPopupRibbon to setPopupState(name: string, state: "open" | "ribbon" | "closed")